test/index.js

```diff
@@ -18,6 +18,8 @@ global.d      = new Date();
 Promise.longStackTraces();
 
 describe('Query Building Tests', function() {
+  this.timeout(process.env.KNEX_TEST_TIMEOUT || 5000);
```

Comment:
Adding beforeEach() outside of describes was ran before every test, but setting timeout there didn't work:

```js
/*global describe*/

'use strict';

require('source-map-support').install();

global.sinon = require("sinon");

var chai = global.chai = require("chai");

chai.use(require("sinon-chai"));
chai.should();

var Promise   = global.testPromise = require('../lib/promise');
global.expect = chai.expect;
global.d      = new Date();

Promise.longStackTraces();

beforeEach(function () {
  console.log("This log is ran before every test, but setting timeout here has no effect!");
  this.timeout(process.env.KNEX_TEST_TIMEOUT || 5000);
})

describe('Query Building Tests', function() {
  require('./unit/query/builder')
  require('./unit/schema/mysql')('mysql')
  require('./unit/schema/mysql')('maria')
  require('./unit/schema/mysql')('mysql2')
  require('./unit/schema/postgres')
  require('./unit/schema/sqlite3')
  require('./unit/schema/oracle')
  require('./unit/schema/mssql')
})

describe('Integration Tests', function() {
  require('./integration')
})
```